packages/happy-dom/test/location/Location.test.ts

```diff
@@ -30,4 +30,12 @@ describe('Location', () => {
 			expect(location.href).toBe(HREF);
 		});
 	});
+
+	describe('href', () => {
+		it('Set href.', () => {
+			location.replace(HREF);
```

Comment:
You should test against a plain string like /Foo instead of using a valid url

Reply:
did you means `location.href = '/foo'` directly, like this:

```javascript
const {Window} = require('happy-dom');
const window = new Window();

window.location.href = '/foo'
```

Reply:
Yes, exactly. I'm watching this from my phone but it seems that HREF is a complete and valid url.

Reply:
> Yes, exactly. I'm watching this from my phone but it seems that HREF is a complete and valid url.


we need have a valid url before we `location.href = '/foo'`, it's same in browser. `/foo` only a path, **it's not valid url**. you can set href `/foo` because it's replace pathname only.
you can test in your browser devtools:

```javascript
location.href = 'about:blank' // the blank page in browser.

location.href = '/foo'

// Uncaught DOMException: Failed to set the 'href' property on 'Location': '/foo' is not a valid URL.
//    at <anonymous>:1:15

```

Reply:
@ilteoood as @Mas0nShi said, the URL needs to be relative to something and by default the URL is set to "about:blank".

Only setting `location.href = '/foo'` will fail with an error as it is not possible to construct a relative URL from "about:blank". Happy DOM could use a different default value for the URL, but what would that be?

After this fix, you will still have to use a setup script for your tests and set a default URL for the URL:s to be relative from.

Reply:
I have added unit tests that demonstrates this.